File: 2022-2023/sourcecode/lab/lab10/MVVMBasicsJava/app/src/main/java/pl/udu/uwr/pum/mvvmbasicsjava/MyViewModel.java
package pl.udu.uwr.pum.mvvmbasicsjava;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MyViewModel extends ViewModel {

    private final MutableLiveData<String> myData = new MutableLiveData<>();

    public LiveData<String> getMyData() {
        return myData;
    }

    public MyViewModel(){
        myData.setValue(getModel().getData());
    }

    private Model getModel() {
        return new Model("text", 1, 2);
    }
}

File: 2022-2023/sourcecode/lab/lab10/MVVMBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/mvvmbasicskotlin/MyViewModel.kt
package pl.udu.uwr.pum.mvvmbasicskotlin

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MyViewModel : ViewModel() {

    private var _myData: MutableLiveData<String> = MutableLiveData()
    val myData: LiveData<String> = _myData

    private fun getModel(): Model {
        return Model("text", 1, 2)
    }

    init {
        _myData.value = getModel().getData()
    }
}